File: tella-locking-ui/src/main/java/com/hzontal/tella_locking_ui/ui/SuccessUpdateDialog.java
package com.hzontal.tella_locking_ui.ui;

import android.app.Dialog;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

import com.hzontal.tella_locking_ui.R;

public class SuccessUpdateDialog extends DialogFragment {

    public static SuccessUpdateDialog newInstance() {
        SuccessUpdateDialog dialog = new SuccessUpdateDialog();
        Bundle bundle = new Bundle();
        dialog.setArguments(bundle);

        return dialog;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // This removes black background below corners.
        setStyle(R.style.DialogTheme,0);
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        // creating the fullscreen dialog
        final Dialog dialog = new Dialog(getActivity());
        dialog.requestWindowFeature(Window.FEATURE_NO_TITLE);
        dialog.getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
        dialog.getWindow().setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);
        return dialog;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.dialog_update_lock_success, container, true);
        getDialog().setCanceledOnTouchOutside(true);
        return view;
    }

    @Override
    public void onStart() {
        super.onStart();
        Dialog dialog = getDialog();
        if (dialog != null) {
            int width = ViewGroup.LayoutParams.MATCH_PARENT;
            int height = ViewGroup.LayoutParams.MATCH_PARENT;
            dialog.getWindow().setLayout(width, height);
        }
    }

}

File: tella-locking-ui/src/main/java/com/hzontal/tella_locking_ui/ui/SuccessUpdateDialog.kt
package com.hzontal.tella_locking_ui.ui

import android.app.Activity
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.hzontal.tella_locking_ui.FINISH_ACTIVITY_REQUEST_CODE
import com.hzontal.tella_locking_ui.R
import com.hzontal.tella_locking_ui.common.CommonStates
import java.util.concurrent.Executors
import java.util.concurrent.TimeUnit

class SuccessUpdateDialog : AppCompatActivity(){
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        overridePendingTransition(R.anim.`in`, R.anim.out)
        finishActivity()
        setContentView(R.layout.dialog_update_lock_success)
    }

    private fun finishActivity(){
        finishActivity(FINISH_ACTIVITY_REQUEST_CODE)
        val executor = Executors.newSingleThreadScheduledExecutor()
        val hideDialog = Runnable {
            CommonStates.finishUpdateActivity.postValue(true)
            setResult(Activity.RESULT_OK)
            finish()
        }
        executor.schedule(hideDialog, 3, TimeUnit.SECONDS);
    }
}